lib/index.js

```diff
@@ -1,239 +1,8 @@
-        if (fullPathModuleId === fullPathItem) {
-          return item;
-        }
-      }
-    }
-
-    return false;
-  }
-
-  verify(options) {
-    let messages = [];
-    let pendingStatus = this.statusForModule('pending', options.moduleId);
-    let shouldIgnore = this.statusForModule('ignore', options.moduleId);
-
-    if (shouldIgnore) {
-      return messages;
-    }
-
-    let source = stripBom(options.source);
-
-    let templateAST;
-
-    try {
-      templateAST = parse(source);
-    } catch (error) {
-      let message = buildErrorMessage(options.moduleId, error);
-      messages.push(message);
-    }
-
-    let rules = this.buildRules({
-      results: messages,
-      pending: pendingStatus,
-      moduleId: options.moduleId,
-      moduleName: options.moduleId,
-      configResolver: options.configResolver,
-      rawSource: source,
-    });
-
-    for (let rule of rules) {
-      try {
-        transform(templateAST, () => rule.getVisitor());
-      } catch (error) {
-        let message = buildErrorMessage(options.moduleId, error);
-        messages.push(message);
-      }
-    }
-
-    if (pendingStatus && messages.length === 0) {
-      messages.push({
-        message: `Pending module (\`${options.moduleId}\`) passes all rules. Please remove \`${options.moduleId}\` from pending list.`,
-        moduleId: options.moduleId,
-        severity: 2,
-      });
-    }
-
-    return messages;
-  }
-
-  logLintingError(name, moduleName, message) {
-    this._messages.push({
-      rule: name,
-      moduleId: moduleName,
-      message,
-    });
-
-    this.console.log(message);
-  }
-
-  static errorsToMessages(filePath, errors, options) {
-    errors = errors || [];
-    options = options || {
-      verbose: false,
-    };
-
-    if (errors.length === 0) {
-      return '';
-    }
-
-    let errorsMessages = errors.map(error => this._formatError(error, options)).join('\n');
-
-    return `${chalk.underline(filePath)}\n${errorsMessages}\n`;
-  }
-
-  static _formatError(error, options) {
-    let message = '';
-
-    let line = error.line === undefined ? '-' : error.line;
-    let column = error.column === undefined ? '-' : error.column;
-
-    message += chalk.dim(`  ${line}:${column}`);
-
-    if (error.severity === WARNING_SEVERITY) {
-      message += `  ${chalk.yellow('warning')}`;
-    } else {
-      message += `  ${chalk.red('error')}`;
-    }
-
-    message += `  ${error.message}  ${chalk.dim(error.rule)}`;
-
-    if (options.verbose) {
-      message += `\n${error.source}`;
-    }
-
-    return message;
-  }
-}
-
-module.exports = Linter;
+module.exports = require('./linter');
 module.exports.Rule = require('./rules/base');
 module.exports.ASTHelpers = require('./helpers/ast-node-info');
 
-module.exports.WARNING_SEVERITY = WARNING_SEVERITY;
-module.exports.ERROR_SEVERITY = ERROR_SEVERITY;
+module.exports.WARNING_SEVERITY = require('./linter').WARNING_SEVERITY;
```

Comment:
FWIW, this mutates the `Linter` class, so we might as well make these things static fields in the Linter file...

Reply:
Interesting 👍.

I've updated the PR with your suggestion.

I wondered what to do with these constants. Having them as static class fields looks good.

Reply:
Unfortunately, static class fields are not supported yet in eslint itself (because it is in stage 3?). So I have to use `babel-eslint` as an eslint parser. Does it look okay?

Reply:
Using static class fields actually requires to add `babel` to the project 😫.

What I've done is refactoring the exports not to mutate the `Linter`. This is explained in the commits. (I'll clean up the history.)

Reply:
eeeck, I was thinking:

```
class Foo {
  static get WARNING_SEVERITY() {
    return WARNING_SEVERITY;
  }
}
```

which should work without transpilation

Reply:
okayy, ah sorry, I will follow up